src/read.rs

```diff
@@ -1489,9 +1631,27 @@ pub(crate) fn parse_single_extra_field<R: Read>(
                 _ => return Err(ZipError::InvalidArchive("Invalid AES vendor version")),
             };
             match aes_mode {
```

Comment:
Eliminate the redundant code here by populating an `AesMode` variable (can probably reuse the name `aes_mode`) first.

Reply:
Done!